test/src/kernels/MapMultiplyCoupledVars.C

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MapMultiplyCoupledVars.h"
+
+registerMooseObject("MooseTestApp", MapMultiplyCoupledVars);
+
+InputParameters
+MapMultiplyCoupledVars::validParams()
+{
+  InputParameters params = ADKernel::validParams();
+  params.addRequiredCoupledVar("v", "The first coupled var");
+  params.addRequiredCoupledVar("w", "The second coupled var");
+  params.addRequiredParam<std::map<std::string, Real>>("coupled_var_multipliers",
+                                                       "A map from variable names to multipliers");
+  return params;
+}
+
+MapMultiplyCoupledVars::MapMultiplyCoupledVars(const InputParameters & parameters)
+  : ADKernel(parameters), _v(adCoupledValue("v")), _w(adCoupledValue("w"))
+{
+  const auto & coupled_map = getParam<std::map<std::string, Real>>("coupled_var_multipliers");
```

Comment:
can we move this line into above header? I want to have a show-case that we can construct a const reference member variable like other parameters.

Reply:
k done